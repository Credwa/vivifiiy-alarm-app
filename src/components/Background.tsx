import React from 'react';
import { ImageBackground, StyleSheet } from 'react-native';

const s3ImageUri = require('~/assets/images/backgroundImage.png');

export default function Background(props: { children: React.ReactNode; paddingHorizontal?: number }) {
  return (
    <ImageBackground
      source={s3ImageUri}
      style={{ ...styles.imageBackground, paddingHorizontal: props.paddingHorizontal || 20 }}
    >
      {props.children}
    </ImageBackground>
  );
}

const styles = StyleSheet.create({
  imageBackground: {
    flex: 1,
    resizeMode: 'cover',
    paddingHorizontal: 20
  }
});
