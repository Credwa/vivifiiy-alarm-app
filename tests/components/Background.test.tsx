import React from 'react';
import renderer from 'react-test-renderer';
import Background from '@/components/Background';

describe('<Background/>', () => {
  it(`renders correctly`, async () => {
    const tree = await renderer.create(<Background>Snapshot test!</Background>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
