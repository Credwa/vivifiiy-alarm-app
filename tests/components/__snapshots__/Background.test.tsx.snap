// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background/> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  Snapshot test!
</View>
`;
